library/core/src/ptr/const_ptr.rs

```diff
@@ -38,32 +38,30 @@ impl<T: ?Sized> *const T {
         self as _
     }
 
-    /// Returns `None` if the pointer is null, or else returns a reference to
+    /// Returns `None` if the pointer is null, or else returns a shared reference to
     /// the value wrapped in `Some`.
     ///
     /// # Safety
     ///
-    /// While this method and its mutable counterpart are useful for
-    /// null-safety, it is important to note that this is still an unsafe
-    /// operation because the returned value could be pointing to invalid
-    /// memory.
-    ///
     /// When calling this method, you have to ensure that *either* the pointer is NULL *or*
     /// all of the following is true:
-    /// - it is properly aligned
-    /// - it must point to an initialized instance of T; in particular, the pointer must be
-    ///   "dereferenceable" in the sense defined [here].
+    ///
+    /// * The pointer must be properly aligned,
+    ///
+    /// * it must be "dereferencable" in the sense defined in [the module documentation], and
+    ///
+    /// * the lifetime `'a` returned is arbitrarily chosen and does
```

Comment:
The wording is odd. You must ensure that the lifetime is arbitrarily chosen? I'd start with something like "*You* must enforce Rust's aliasing rules, since the returned lifetime `'a` is arbitrarily chosen. In particular, ..."